src/containers/GameContainer/GameContainer.tsx

```diff
@@ -37,73 +75,107 @@ export const GameContainer = () => {
-    });
-
-    canvas.addEventListener('mousedown', () => {
-      mainHero.startBoost();
-    });
-    canvas.addEventListener('mouseup', () => {
-      mainHero.endBoost();
-    });
-
-    game.start();
+    game.beforeStart = () => {
+      setGameStatus('inGame');
+      timeController.start();
+    };
+
+    game.logic = (timeFraction: number) => {
+      timeController.updateTime(timeFraction);
+
+      if (timeController.allowedMove) {
+        borderGame.color = config.GAME_BORDER_ALLOWED_COLOR;
+      }
+
+      if (timeController.safePeriod) {
+        borderGame.color = config.GAME_BORDER_WARNING_COLOR;
+      }
+
+      if (timeController.timeout) {
+        borderGame.color = config.GAME_BORDER_STOP_COLOR;
+      }
+
+      if (mainHero.leftBorder > finishLine.lengthDistance) {
+        mainHero.isWin = true;
+      }
+
+      if (mainHero.speed > 0 && timeController.timeout && !mainHero.isWin) {
+        mainHero.isLost = true;
+      }
+
+      if (timeController.endTime && !mainHero.isWin) {
+        borderGame.color = config.GAME_BORDER_STOP_COLOR;
+        mainHero.isLost = true;
+      }
+
+      if (mainHero.isLost) {
+        timeController.stop();
+        setGameStatus('lose');
+      }
+
+      if (mainHero.isWin) {
+        timeController.stop();
+        setGameStatus('win');
+      }
+
+      if (timeController.timeOver && !mainHero.isLost && !mainHero.isWin) {
+        timeController.reset();
+      }
+
+      mainHero.move(timeFraction);
+      mainHero.move(timeFraction);
+    };
+
+    game.render = () => {
+      borderGame.render(ctx);
+      mainHero.render(ctx);
+      finishLine.render(ctx);
+      const tTime = new Date(timeController.remainingTime);
+
+      setRenderTime(`${tTime.getMinutes()}:${tTime.getSeconds()}`);
+    };
+
+    game.clear = () => ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+    game.restart = () => {
+      mainHero.refresh();
+      timeController.refresh();
+      setGameStatus('inGame');
+    };
+
+    return () => {
+      game.clearAnimate();
+    };
   }, []);
 
-  return <canvas ref={ref} width={config.GAME_WIDTH} height={config.GAME_HEIGHT} />;
+  const startGame = () => {
+    game.start();
+  };
+  const restartGame = () => {
+    game.restart();
+  };
+  const startBoost = () => {
+    mainHero.startBoost();
+  };
+  const endBoost = () => {
+    mainHero.endBoost();
+  };
+
+  return (
+    <div onMouseDown={startBoost} onMouseUp={endBoost}>
+      <div>{renderTime}</div>
+      <canvas ref={canvasRef} width={config.GAME_WIDTH} height={config.GAME_HEIGHT} />
+      {gameStatus !== 'inGame' && (
+        <ModalWrapper>
+          {gameStatus === 'beforeGame' ? (
+            <StartWindow startGame={startGame} />
+          ) : gameStatus === 'lose' ? (
+            <LoseWindow restartGame={restartGame} />
+          ) : gameStatus === 'win' ? (
+            <WinWindow restartGame={restartGame} />
+          ) : null}
```

Comment:
🧐
Подобные условия в JSX лучше выглядят вот так. Возможно, субъективщина, поэтому не ❗️, но кмк это лучше читается
```suggestion
          {gameStatus === 'beforeGame' && (
            <StartWindow startGame={startGame} />
          )}
          {gameStatus === 'lose' && (
            <LoseWindow restartGame={restartGame} />
          )}
          {gameStatus === 'win' && (
            <WinWindow restartGame={restartGame} />
          )}
```